airflow/models/dag.py

```diff
@@ -2880,12 +2883,16 @@ def dags_needing_dagruns(cls, session: Session):
         # TODO[HA]: Bake this query, it is run _A lot_
         # We limit so that _one_ scheduler doesn't try to do all the creation
         # of dag runs
+        # exclude dags with import error
+        import_errors = [c.filename for c in session.query(importerror.filename).all()]
         query = (
             session.query(cls)
             .filter(
+                cls.max_active_runs != None,  # noqa
```

Comment:
What does this have to do with import errors?

Reply:
Actually the cause of the issue was that an active dag had errors after upgrade, normally it should have max_active_runs but it didn’t have it because it failed to parse. 

I’ll update it to only include this check and remove import errors